feat: Update Project

euny0ung opened the pull request:
프로젝트 페이지를 구현했습니다.
연구분야 페이지와 비슷하게 만들었습니다. 연구분야 페이지 코드에서 상단의 설명 박스와 scrolling pallarax 기능을 제거하였습니다.

![프로젝트 페이지 완성](https://user-images.githubusercontent.com/56199780/169832115-ea17fe5a-a28a-4581-9e0b-e4e15694158a.PNG)

euny0ung committed:
feat: Update Project

프로젝트 페이지를 구현했습니다.

#43

suinj8 reviewed (commented):
이유는 모르겠지만 sass모듈이 의존성에 추가되어있지 않은것 같네요.
npm i를 해도 설치되지 않습니다.
저는 따로 npm install sass해서 실행시켜 봤네요. 

일단은 LGTM

LimEunSang reviewed (changes requested):
연구 분야 페이지에서 작성한 comment와 마찬가지로 각 프로젝트 분야의 내용을 하나의 컴포넌트로 구현하면 좋지 않을까 의견을 제시합니다.

또한 sass 모듈에 대한 의존성이 추가돼있지 않아 따로 npm i sass 해줘야 실행이 되네요.
제가 직접 의존성 추가해서 commit 할 수도 있는데 저번에 수인이한테 commit 내역이 꼬여서 고생하셨다는 애기 듣고 왠만하면 상대방 코드 수정 안하는게 나을 거 같아요. 의존성 추가해주세요.

euny0ung committed:
nothing

euny0ung committed:
nothing

euny0ung committed:
fix: Resolve conflict
충돌 해결

#44

euny0ung committed:
Merge branch 'feature/project' of https://github.com/AI-Research-Center-Homepage/frontend into feature/project

euny0ung committed:
refactor: Update Project
-버튼 이벤트 구현방식을 바꿨습니다
-데이터를 Object로 묶어 컴포넌트 하나를 재사용하는 방식으로 바꿨습니다.
- useState의 초기값을 첫번째 버튼값으로 하여 기본으로 첫번째 화면이 뜨게 하였습니다.

#44

euny0ung committed:
Merge branch 'feature/project' of https://github.com/AI-Research-Center-Homepage/frontend into feature/project

euny0ung commented:
의존성 추가하였습니다. 잘 되는지 확인 부탁드려요.
말해주신 부분들 다 수정하였습니다. 그런데 연구분야 페이지와 마찬가지로 화면출력이 됐다가 안됐다가 하는 문제가 있네요(46번 이슈)

euny0ung committed:
fix: Resolve problem
화면 출력이 랜덤으로 되는 문제 해결
-Typography 태그 삭제

#44

suinj8 reviewed (commented):
큰따옴표와 작은따옴표의 통일이 필요해 보입니다.
크게 상관은 없지만 통일성이 조금 떨어져 보이네요.

euny0ung commented:
> 큰따옴표와 작은따옴표의 통일이 필요해 보입니다. 크게 상관은 없지만 통일성이 조금 떨어져 보이네요.

작은 따옴표 검색하면 여기만 나오는데 혹시 이거 말하신거 맞나요??
<img width="338" alt="따옴표" src="https://user-images.githubusercontent.com/56199780/171022092-36776937-9996-46f1-aeeb-fd985517e2d9.PNG">

LimEunSang reviewed (approved):
제 개인적인 생각입니다!
컴포넌트화 하는 목적은 재사용이잖아요, 바꿔 말하면 재사용하는 코드는 컴포넌트화를 해야합니다. 그래야 개발시에도 시간이 단축되고 유지보수 측면에서도 상당히 효율적이게 되니까요.

프로젝트 페이지는 코드가 꽤 긴데 거의 모든 부분이 연구분야 페이지랑 동일합니다. 이 부분도 컴포넌트화를 더 진행할 수 있을 거 같아요.
그리고 Description 컴포넌트 잘 정의해놨는데, 따로 파일로 만들어주지 않아서 결국 두 페이지에서 똑같은 컴포넌트를 정의하고 있어요.

제가 다른 건 몰라도 개발에서 컴포넌트화(함수화, 객체화)가 중요하다고 귀에 피가 나도록 들어서 집착하게 되네요..

이런 부분들은 병합하고 나서 제가 능력껏 수정해볼게요. 이후에 은영님이 컴포넌트화를 하는게 효율적이고 중요하다고 느껴지면, 관리자 홈페이지를 만들때 신경써서 진행해주시면 좋을거같아요.

페이지 구현하느라 고생하셨습니다! LGTM!

suinj8 commented:
![image](https://user-images.githubusercontent.com/90179555/171109911-d634497d-2f88-42e1-9a08-42bdb91c2b8d.png)




> > 큰따옴표와 작은따옴표의 통일이 필요해 보입니다. 크게 상관은 없지만 통일성이 조금 떨어져 보이네요.
> 
> 작은 따옴표 검색하면 여기만 나오는데 혹시 이거 말하신거 맞나요?? <img alt="따옴표" width="338" src="https://user-images.githubusercontent.com/56199780/171022092-36776937-9996-46f1-aeeb-fd985517e2d9.PNG">

App.js , Professor.js 파일을 보시면 작은따옴표로 표기되어 있습니다. 
확인 부탁드립니다.

euny0ung committed:
style: Update App.js
작은 따옴표를 큰 따옴표로 변경

#44

euny0ung committed:
refactor: Update App.js , Project.js
주석 띄어쓰기 및 구간 나누기

#44

euny0ung commented:
Project 파일 주석 띄어쓰기, 구간 나누기 완료하였습니다. 구간 나누기의 경우 줄바꿈을 해도 prettier 때문에 안먹히는(?) 부분이 있어서 그런건 그냥 두었습니다. ex) ResearchField의 303번째 줄

suinj8 reviewed (approved):
확인하였습니다.
LGTM!

baaaam771 reviewed (approved):
연구분야 페이지랑 유사하네요 고생하셨습니다 LGTM

euny0ung committed:
Merge branch 'develop' into feature/project
